pkg/scannerv4/mappers/mappers.go

```diff
@@ -386,10 +379,58 @@ func toProtoV4VulnerabilitiesMap(ctx context.Context, vulns map[string]*claircor
 			FixedInVersion:     fixedInVersion(v),
 			Cvss:               cvss,
 		}
+		if len(cvssMetrics) > 0 {
+			vuln.CvssMetrics = cvssMetrics
+		}
+		if vulnerabilities == nil {
+			vulnerabilities = make(map[string]*v4.VulnerabilityReport_Vulnerability, len(vulns))
+		}
+		vulnerabilities[k] = vuln
 	}
 	return vulnerabilities, nil
 }
 
+// toCVSSMetrics adds a CVSS metric, based on the provided severity values, to the list of CVSS metrics in the v4 vulnerability report.
+func toCVSSMetrics(ctx context.Context, sev severityValues, cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS) []*v4.VulnerabilityReport_Vulnerability_CVSS {
+	cvss, err := toCVSSMetric(sev)
+	if err != nil {
+		zlog.Warn(ctx).
+			Err(err).
+			Str("source", sev.source.String()).
+			Msg("No CVSS Metric")
+		return cvssMetrics
+	}
+	cvss.Source = sev.source
+	cvssMetrics = append(cvssMetrics, cvss)
+	return cvssMetrics
+}
```

Comment:
This doesn't seem very necessary. I feel like `toCVEEMetric` can set the `Source` and the caller can handle errors. Is this because it's called again for the older `cvss` field? If so, I think it's ok we add `Source` to that. If Scanner V4's client can't handle that, it's ok, as protobufs are backwards compatible (the client will just ignore it)

Reply:
I changed toCVSSMetrics() to convert sev slice to CVSS metrics